sympy/sets/fancysets.py

```diff
@@ -544,27 +561,32 @@ def __new__(cls, *args):
     Either the start or end value of the Range must be finite.'''))
 
         if start.is_infinite:
-            if step*(stop - start) < 0:
-                start = stop = S.One
+            if Lt(step*(stop - start), 0) == True:
```

Comment:
I don't know why you're using `Lt` rather than `<`. If you want to be direct then just use:
```
    if (step*(stop - start)).is_extended_negative:
```
(Since that is the condition that Lt will use to evaluate)